### PR TITLE
Fix VaultPress query refresh

### PR DESCRIPTION
**Pull request by mreishus**
#### Changes proposed in this Pull Request:

* Fixed a bug that caused endless fetching of VaultPress data in some circumstances, causing cards on the jetpack dashboard to flicker

https://user-images.githubusercontent.com/937354/166066158-b3cc10c5-10ac-40b3-8b70-cb689426794a.mp4

* I have added a root cause analysis of the bug later in the comments in this PR: https://github.com/Automattic/jetpack/pull/24194#issuecomment-1113840687

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Have a Jetpack site without this PR
* Install and activate the VaultPress plugin, but don't register
* Visit the JP Dashboard (`/wp-admin/admin.php?page=jetpack#/dashboard`)
* Note that some cards are flickering
* Repeat steps with this PR; this time, the flickering should not be present
* VaultPress related functionality in Jetpack should continue to work
  * ^ The above is vague because I know very little about it.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/state/at-a-glance/reducer.js [hunk 1]
```diff
@@ -165,8 +165,8 @@ export const dashboard = combineReducers( {
 /**
  * Returns string of active Stats tab in At A Glance section
  *
- * @param  {Object}  state  Global state tree
- * @return {String}         Which Stats tab is open.
+ * @param  {Object}  state -  Global state tree
+ * @returns {string}         Which Stats tab is open.
```
Comment (mreishus) [inline projects/plugins/jetpack/_inc/client/state/at-a-glance/reducer.js:169]: These changes are done by prettier and automatically fix errors like:
```
/home/matta8c/h21/dev/jetpack/projects/plugins/jetpack/_inc/client/state/at-a-glance/reducer.js                                                             
  255:1  warning  Missing JSDoc @returns declaration                                         jsdoc/require-returns                                          
  257:0  warning  There must be no indentation                                               jsdoc/check-indentation                                        
  259:0  warning  There must be a hyphen before @param description                           jsdoc/require-hyphen-before-param-description                  
  260:0  warning  Invalid JSDoc tag (preference). Replace "return" JSDoc tag with "returns"  jsdoc/check-tag-names                                          
  260:0  warning  Invalid JSDoc @return type "Boolean"; prefer: "boolean"                    jsdoc/check-types                                              
```

**Comment by mreishus** [comment]
The biggest risk with this is stale vaultpress data. Before this PR, remounting the query component refetches the data, but now there's no way to force the refetch, which may be needed. :thinking:

**Comment by mreishus** [comment]
I was looking into the root cause, to see if there's a way to fix this by stopping `<QueryVaultPressData />` from remounting, instead of changing what happens when it remounts.

The chain of events is: 

1. `<QueryVaultPressData />` mounts and fires off a request for vaultPressData, which sets `fetchingVaultPressData` to be true.
2. `<DashScan />` sees that we are `fetchingVaultPressData` and stops rendering its children.
https://github.com/Automattic/jetpack/blob/337a1b68d22bcc9572c9f2cc63c673e23b76d1ce/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx#L433-L438
3. The chain of children components for scan, `<DashItem />` -> `<ProStatus />` -> `<QueryVaultPressData />` all **unmount**.
4. The fetch for vaultpress data resolves and `fetchingVaultPressData` is set to false.
5. `<DashScan />` sees that we are no longer  `fetchingVaultPressData`  and starts rendering its children.
https://github.com/Automattic/jetpack/blob/337a1b68d22bcc9572c9f2cc63c673e23b76d1ce/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx#L433-L438
6. The chain of children components for scan, `<DashItem />` -> `<ProStatus />` -> `<QueryVaultPressData />` all **mount**.
7. Go to step 1.

Something like this also fixes the issue:
```diff
diff --git a/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx b/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx
index 2b1bb6114c..1defa761c0 100644
--- a/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx
+++ b/projects/plugins/jetpack/_inc/client/at-a-glance/scan.jsx
@@ -425,17 +425,7 @@ class DashScan extends Component {
                        } );
                }
 
-               // Wait until we know everything about the site's VaultPress/Rewind scan
-               // status before showing any "real" content
-
-               // ASSUMPTION: QueryVaultPressData and QueryScanStatus have been invoked
-               //             elsewhere on the page.
-               const isLoading = this.props.fetchingScanStatus || this.props.fetchingVaultPressData;
-               return (
-                       <div>
-                               { isLoading ? renderCard( { content: __( 'Loading…', 'jetpack' ) } ) : this.getContent() }
-                       </div>
-               );
+               return <div>{this.getContent()}</div>;
        }
 }
 ```

But is probably not something we want, as the loading behavior was added in #17288.

**Review by mattgawarecki — APPROVED** [review]
I haven't functionally tested, but the changes in this PR make sense to me from the perspective of fixing an infinite re-render loop, so I'm adding my approval.

> The biggest risk with this is stale vaultpress data.

Do we know how much of an issue this will cause? Intuitively I would guess that every fresh visit to the Jetpack section of wp-admin would re-fetch data, but navigation _within_ the section would not. Does VaultPress data change often enough to possibly confuse folks with an incorrect status?

**Comment by mreishus** [comment]
> Do we know how much of an issue this will cause? Intuitively I would guess that every fresh visit to the Jetpack section of wp-admin would re-fetch data, but navigation _within_ the section would not. Does VaultPress data change often enough to possibly confuse folks with an incorrect status?

Unfortunately, I haven't found a way to have vaultpress data be anything other than "N/A". Some more about what I've tried in #24193

**Comment by mreishus** [comment]
@obenland found a way to get vpData, but it requires a JN site

**Comment by mreishus** [comment]
Tried it out, looks good. I agree that stale data is less of an issue than an infinite loop which essentially DOSes the endpoint. Looks like I need an official Jetpack reviewer before it can be merged though:

**Review by mmtr — APPROVED** [review]
Looks good! I had to a Scan Daily product first to reproduce the flickering, which this PR is preventing from happening. 🚀

**Pull request merged**